erroranalysis/erroranalysis/_internal/surrogate_error_tree.py

```diff
@@ -326,6 +328,15 @@ def node_to_dict(df, tree, nodeid, categories, json,
     if parent is not None:
         parentid = int(parent[SPLIT_INDEX])
         p_node_name = feature_names[parent[SPLIT_FEATURE]]
+        # use number.Integral to check for any numpy or python number type
+        if isinstance(p_node_name, numbers.Integral):
+            # for numeric column names, we can use @df[numeric_colname] syntax
+            p_node_query = "@df[" + str(p_node_name) + "]"
+        else:
+            # for string column names, we can just use column name directly
+            # with backticks
+            p_node_query = "`" + str(p_node_name) + "`"
+        p_node_name = str(p_node_name)
```

Comment:
IS this something we should support? We could also catch this condition and raise a user error if we think that is something a user can fix in their data.

Reply:
yes, we definitely should, it's valid to have numeric feature names - both this customer and actually Ke at some point earlier hit this issue.  I don't think we should throw (that's what we did earlier) since it would block customers from running a valid scenario?  What do you think?

Reply:
Sure sounds good.